### PR TITLE
[2.8.6] added doc for Rancher CLI login with AzureAD

### PR DESCRIPTION
**Pull request by enrichman**
This PR adds some docs related to the new feature about the authentication from the Rancher CLI with the AzureAD provider.

Ref:
- https://github.com/rancher/rancher/issues/42939
- https://github.com/rancher/rancher/issues/44282

The issues are still in "To Test", so I would wait a validation from QA before merging this PR into the docs.

### CODE REVIEW

File: docs/how-to-guides/new-user-guides/authentication-permissions-and-global-configuration/authentication-config/configure-azure-ad.md [hunk 1]
```diff
@@ -133,7 +133,17 @@ Here are a few examples of permission combinations that satisfy Rancher's needs:
 
 :::
 
-#### 4. Copy Azure Application Data
+#### 4. Allow Public Client Flows
+
+To login from Rancher CLI you will need to allow Public Client Flows.
```
Comment (crobby) [inline docs/how-to-guides/new-user-guides/authentication-permissions-and-global-configuration/authentication-config/configure-azure-ad.md:138]: Per the check, the word "will" should be removed.

Comment (enrichman) [reply docs/how-to-guides/new-user-guides/authentication-permissions-and-global-configuration/authentication-config/configure-azure-ad.md:138]: Thanks @crobby, before fixing it, is the rest ok? :D

Comment (martyav) [inline docs/how-to-guides/new-user-guides/authentication-permissions-and-global-configuration/authentication-config/configure-azure-ad.md:138]:
```suggestion
To login from Rancher CLI you must allow Public Client Flows.
```

File: docs/how-to-guides/new-user-guides/authentication-permissions-and-global-configuration/authentication-config/configure-azure-ad.md [hunk 2]
```diff
@@ -133,7 +133,17 @@ Here are a few examples of permission combinations that satisfy Rancher's needs:
...
+
+1. From the navigation pane on, select **Authentication**.
```
Comment (martyav) [inline docs/how-to-guides/new-user-guides/authentication-permissions-and-global-configuration/authentication-config/configure-azure-ad.md:140]:
```suggestion
1. From the left navigation menu, select **Authentication**.
```

File: docs/how-to-guides/new-user-guides/authentication-permissions-and-global-configuration/authentication-config/configure-azure-ad.md [hunk 3]
```diff
@@ -133,7 +133,17 @@ Here are a few examples of permission combinations that satisfy Rancher's needs:
...
+
+1. In the **Advanced Settings** enable the **public client flows**.
```
Comment (martyav) [inline docs/how-to-guides/new-user-guides/authentication-permissions-and-global-configuration/authentication-config/configure-azure-ad.md:142]:
```suggestion
1. Under **Advanced Settings**, enable **public client flows**.
```

**Review by martyav — CHANGES_REQUESTED** [review]

File: docs/how-to-guides/new-user-guides/authentication-permissions-and-global-configuration/authentication-config/configure-azure-ad.md (hunk 2 above)
Comment (enrichman) [reply docs/how-to-guides/new-user-guides/authentication-permissions-and-global-configuration/authentication-config/configure-azure-ad.md:140]: Thanks @martyav. I just want to highlight that this was copy-pasted from line 104. Should I update also that?

**Review by martyav — APPROVED** [review]

**Review by crobby — DISMISSED** [review]

**Comment by martyav** [comment]
As noted above, 

> The issues are still in "To Test", so I would wait a validation from QA before merging this PR into the docs.

**Comment by enrichman** [comment]
> As noted above,
> 
> > The issues are still in "To Test", so I would wait a validation from QA before merging this PR into the docs.
> 
> The related issues are for v2.9-next1

Thank you @martyav. Since this was backported to 2.8 it will be present also in the next `v2.8.6` release (see https://github.com/rancher/rancher/issues/44282 with the 2.8-next1 milestone).

I don't know if something else needs to be done.

**Comment by martyav** [comment]
> > As noted above,
> > > The issues are still in "To Test", so I would wait a validation from QA before merging this PR into the docs.
> > 
> > 
> > The related issues are for v2.9-next1
> 
> Thank you @martyav. Since this was backported to 2.8 it will be present also in the next `v2.8.6` release (see [rancher/rancher#44282](https://github.com/rancher/rancher/issues/44282) with the 2.8-next1 milestone).
> 
> I don't know if something else needs to be done.

@enrichman Oops, I assumed the two had the same milestone. I'll update the PR milestone and my previous comment.

**Review by martyav — APPROVED** [review]

**Review by martyav — DISMISSED** [review]

**Review by sunilarjun — DISMISSED** [review]

**Review by crobby — APPROVED** [review]

**Comment by enrichman** [comment]
@crobby @sunilarjun @martyav 

While testing again and checking the docs I've found one more place where I had to update this info.

Added with 89d948413613b4b83c5c0d66eb26549329e572b2, and dismissed the approvals.

**Review by crobby — APPROVED** [review]

File: versioned_docs/version-2.9/how-to-guides/new-user-guides/authentication-permissions-and-global-configuration/authentication-config/configure-azure-ad.md [hunk 4]
```diff
@@ -133,7 +133,17 @@ Here are a few examples of permission combinations that satisfy Rancher's needs:
 
 :::
 
-#### 4. Copy Azure Application Data
+#### 4. Allow Public Client Flows
+
+To login from Rancher CLI you must allow public client flows.
```
Comment (martyav) [inline versioned_docs/version-2.9/how-to-guides/new-user-guides/authentication-permissions-and-global-configuration/authentication-config/configure-azure-ad.md:138]:
```suggestion
To login from Rancher CLI you must allow public client flows:
```

**Review by martyav — CHANGES_REQUESTED** [review]
Extremely minor changes requested. Otherwise, LGTM

**Review by sunilarjun — COMMENTED** [review]
Minus the changes mentioned above, looks good to me, thank you!

**Comment by enrichman** [comment]
Thank you all! Applied suggestions from @martyav.

We had the validation from QA, so we are good to go on these.  
Please be aware that we need to update the docs for both the `2.8.6` and the `2.9.0` release!

QA OK:
- https://github.com/rancher/rancher/issues/42939#issuecomment-2226139159
- https://github.com/rancher/rancher/issues/44282#issuecomment-2226145985

**Review by crobby — APPROVED** [review]

**Review by martyav — APPROVED** [review]

**Review by btat — APPROVED** [review]

**Review by martyav — APPROVED** [review]

**Pull request merged**